app/services/item_change_set_persister.rb

```diff
@@ -65,14 +66,12 @@ def updated_access(updated)
   end
 
   def update_identification(updated)
-    updated_identification = updated.identification
-    updated_identification = updated_identification.new(sourceId: source_id) if source_id_changed?
+    identification_props = updated.identification&.to_h || {}
+    identification_props[:sourceId] = source_id if source_id_changed?
+    identification_props[:barcode] = barcode if barcode_changed?
     if catkey_changed?
-      catalog_links = []
-      catalog_links << { catalog: 'symphony', catalogRecordId: catkey } if catkey
-      updated_identification = updated_identification.new(catalogLinks: catalog_links)
+      identification_props[:catalogLinks] = catkey ? [{ catalog: 'symphony', catalogRecordId: catkey }] : nil
     end
-
-    updated.new(identification: updated_identification)
+    updated.new(identification: identification_props.compact.presence)
```

Comment:
Can you say more about this refactoring? This looks related to a conversation we had on the bulk license action PR, but maybe not.

Reply:
`catalogLinks: nil` (which is dropped in compact) instead of `catalogLinks: []`, which is not.

While logically equivalent, this causes problems whenever we try to compare cocina objects. (I think this is going to cause us problems down the road with cocina.)